devon4j-http-rest-client/index.asciidoc

```diff
@@ -0,0 +1,126 @@
+====
+
+
+====
+You have successfully built the REST-server. Now, you have to start the build and then, start the server as mentioned below.
+[step]
+--
+buildJava("httprestserver", false)
+runServerJava("httprestserver/server", { "startupTime": 1000, "port": 8080, "path": "httprestserver" })
+nextKatacodaStep(" ", [{ "file": "files/viewPort.txt" }])
+--
+In next step, You have to create devon4j service Client.
+====
+
+
+====
+To create devon4j service Client. First, You need to create a sample devon4j project.
+
+[step]
+== Create devon4j service Client
+--
+createDevon4jProject("com.sample.application.httprestclient")
+--
+Then, You need to add required dependency to your application. 
+====
+
+
+[step]
+--
+nextKatacodaStep("Dependency Setup ", [{ "file": "files/dependency.txt" }])
+--
+
+====
+You have to create a Java interface VisitormanagementRestService to invoke.
+You have to create rest client i.e. Devon4jRestClient.java and it's implementation class i.e. Devon4jRestClientImpl.java.
+[step]
+== Create devon4j rest client
+--
+createFile("httprestclient/api/src/main/java/com/sample/application/httprestclient/general/service/api/rest/VisitormanagementRestService.java", "files/VisitorManagementRestServiceClient.txt")
+createFile("httprestclient/api/src/main/java/com/sample/application/httprestclient/general/service/api/rest/Devon4jRestClient.java", "files/Devon4jRestClient.java")
+createFile("httprestclient/core/src/main/java/com/sample/application/httprestclient/general/service/impl/rest/Devon4jRestClientImpl.java", "files/Devon4jRestClientImpl.java")
+nextKatacodaStep(" ", [{ "file": "files/explainSynchronousCall.txt" }])
+--
+After getting response, you can handle the response further in your implementation. Here, you can see that response is getting handled in "showResponse" method.
+====
+
+
+====
+# Configuration 
+An application needs to be configurable in order to allow internal setup but also to allow externalized configuration of a deployed package (e.g. integration into runtime environment). 
+Now, You need to modifiy the content of existing properties files and add configuration for Server.
+[step]
+--
+changeFile("httprestclient/core/src/main/resources/application.properties", { "file": "files/client_application.txt" })
+changeFile("httprestclient/core/src/main/resources/config/application.properties", { "file": "files/client_config_application.txt" })
+nextKatacodaStep("Properties Configuration", [{ "file": "files/configurationProperties.txt" }])
+--
+Now, after configuration you have to build and then start the client application.
+====
+
+====
+You have successfully built the devon4j rest client.
+Let's build and start the service client application.
+This might take some time for application to start.
+[step]
+--
+buildJava("httprestclient", false)
+nextKatacodaStep(" ", [{ "file": "files/openTerminalForClient.txt" }])

```

Comment:
You can't pass an empty title right now, this will maybe be added. Right now you have to pass a useful title or use one of the step description to print the text from openTerminalForClient.txt

Reply:
@SchettlerKoehler

Reply:
I have added the title for the step.